src/txn/txn_ckpt.c

```diff
@@ -1413,6 +1413,13 @@ __txn_checkpoint_wrapper(WT_SESSION_IMPL *session, const char *cfg[])
     WT_STAT_CONN_SET(session, txn_checkpoint_running, 1);
     txn_global->checkpoint_running = true;
 
+    /*
+     * FIXME-WT-11149: Some reading threads rely on the value of checkpoint running flag being
+     * published before the checkpoint generation number (set inside the checkpoint call below).
+     * Introduce a write barrier here to guarantee the right order.
+     */
+    WT_WRITE_BARRIER();
```

Comment:
I think to make things clear. It is better to move the this code inside to __txn_checkpoint?

Reply:
I would like to have the opinion of others on this one. This means I would add the following in `__txn_checkpoint`:
```
    WT_STAT_CONN_SET(session, txn_checkpoint_running, 1);
    txn_global->checkpoint_running = true;
    WT_WRITE_BARRIER();
```
And
```
    WT_STAT_CONN_SET(session, txn_checkpoint_running, 0);
    txn_global->checkpoint_running = false;
```
The way `__txn_checkpoint` is implemented makes this tricky when setting `checkpoint_running` to `false` as sometimes we go to the `err` label and sometimes we do `return (0)`. I could create a `done` label and instead of returning 0, we would jump to this label and I could add the following code:
```
    WT_STAT_CONN_SET(session, txn_checkpoint_running, 0);
    txn_global->checkpoint_running = false;
```
I wish there was a tag-all option - I will ping the first three reviewers for now @sueloverso @kommiharibabu @korteland, do you have any thoughts on this?

Reply:
The function call is a compiler barrier. I proposed to move this inside the function call is that there is a dependency  here. If the logic are together, it would make it easier to deduce this dependency.

Reply:
I also feel that keeping the change at the current location would be good rather than moving it inside.